providers/flagd/README.md

```diff
@@ -22,6 +22,7 @@ flagd.WithoutCache()                    // disables caching of flag evaluations
 flagd.WithLRUCache(1000)                // enables LRU caching (see configuring caching section)
 flagd.WithBasicInMemoryCache()          // enables basic in memory cache (see configuring caching section)
 flagd.WithLogger(logger)                // sets a custom logger (see logging section)
+flagd.WithOtelInterceptor(bool)         // enable or disable OpenTelemetry interceptor for flagd communication
```

Comment:
Interceptor uses no-op implementations and yields nothing

The idea here was to allow sdk users to manually opt-in for telemetry interception. Because this is only needed if the sdk user (i.e- application developer) has a proper telemetry configuration for the application. Setup includes exporter configuration (ex:- console, otel exporter ...) and related services up and running (ex:- otel collector). 

So for simple usage as well as to exclude the performance impact of including no-op interceptors, it's best to make this an opt-in configuration.